fs/zbd_zenfs.h

```diff
@@ -94,12 +95,15 @@ class ZonedBlockDevice {
   unsigned int max_nr_active_io_zones_;
   unsigned int max_nr_open_io_zones_;
 
+  std::shared_ptr<ZenFSMetrics> metrics_;
+
   void EncodeJsonZone(std::ostream &json_stream,
                       const std::vector<Zone *> zones);
 
  public:
   explicit ZonedBlockDevice(std::string bdevname,
-                            std::shared_ptr<Logger> logger);
+                            std::shared_ptr<Logger> logger,
+                            std::shared_ptr<ZenFSMetrics> metrics = nullptr);
```

Comment:
Not sure, but I think that when you deref a shared_ptr that holds a nullptr, you get a segfault. We should probably initialize to `std::make_shared<NoZenFSMetrics>()` instead.

Reply:
Yes, this is a serious error. Modified.